src/perform_step/bdf_perform_step.jl

```diff
@@ -398,27 +409,44 @@ function perform_step!(integrator,cache::QNDF1Cache,repeat_step=false)
     κ = zero(alg.kappa)
   end
 
-  # precalculations
-  γ₁ = 1//1
-  nlsolver.γ = γ = inv((1-κ)*γ₁)
-  @.. tmp = uprev + D[1] - γ * (γ₁*D[1])
+  #Changing uprev2 after D Array has changed with step-size
+  uprev2 = uprev - D[1]
+    
+  β₀ = 1 
+  α₀ = 1-κ
+  α₁ = 1-2*κ
+  α₂ = κ
 
   markfirststage!(nlsolver)
 
   # initial guess
-  @.. z = (uprev + D[1] - nlsolver.tmp)*inv(γ)
+  nlsolver.z = uprev + sum(D)
+
+  mass_matrix = f.mass_matrix
+
+  if mass_matrix === I
+    @.. tmp = @.. (α₁ * uprev + α₂ * uprev2) / (dt * β₀)
```

Comment:
There’s a duplicate of @.. Just remember to change it in the next PR to save some CI time.